airflow/providers/elasticsearch/log/es_task_handler.py

```diff
@@ -187,8 +188,18 @@ def _group_logs_by_host(self, logs):
     def _read_grouped_logs(self):
         return True
 
+    @cached_property
+    def supports_triggerer(self):
+        """Not implemented yet."""
+        return False
```

Comment:
Would it be a good idea to implement this as a class variable instead? A property returning a constant seems awkward.

Reply:
yes it's absolutely awkward. mostly this property is taken care of through inheritance from FTH, but, in this case and in wasb, i had to implement the property because, since they used type annotations on _read, mypy complained about inconsistent signature so i had to add the new kwarg and manually set to not supported (the property looks at signature to tell)

Reply:
but the important point @uranusjr is.... 

if we do class variable, it propagates through inheritance... and this is a no-no because subclasses need to be updated to properly work with triggers.  not huge updates in most cases. and the property on FTH is defined to be false when not yet implemented (excepti my beautiful plan fails when mypy complains)

Reply:
Don’t properties also propagate through inheritance? I’m obviously missing something…

Reply:
if the property looks at the signature of the instance method, which has not yet been updated...

Reply:
look at the implementation in FTH, the base class

Reply:
self always refers to instance, so subclasses evaluate it against their methods, not those of the super

Reply:
perhaps you mean that the subclass should implement as attr even though the parent implements as property?

Reply:
So to sum up

1. Mypy complains if we make this a class var in the parent class but property in children
2. We need this to be a property in children because some of those need runtime evaluation

In that case I guess a property is alright. I wish Mypy can improve on this; it’s actually a quite common pattern. (To achieve this we need a way to declare an _attribute_ as read-only, I digress…)

Reply:
Not quite. Mypy complaints about the signature of _read in the subclass not matching that of the parent because I updated the signature of the parent. This left me with the choice of either removing type annotations from the subclass method or updating the subclass method signature even though I am deferring the act of actually updating the subclass to work with triggers. But we use, in this property, inspection of this methods signature to determine whether the subclass supports triggers. Therefore, it would falsely indicate that the subclass does support triggers ( because the subclass signature now has been updated ), therefore we have to implement the property and return false. I did not try just adding it as an attribute because the parent class is a property and I thought it best to stay consistent with respect to that. When the subclass is ultimately updated to support triggers, then we can remove this property from the subclass because the parent class property will then tell the truth.